client/src/test/java/uk/co/blackpepper/bowman/LinkedResourceMethodHandlerTest.java

```diff
@@ -0,0 +1,88 @@
+package uk.co.blackpepper.bowman;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import org.junit.Test;
+import org.springframework.hateoas.Resource;
+
+import uk.co.blackpepper.bowman.annotation.LinkedResource;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.junit.Assert.assertThat;
+import static org.springframework.util.ReflectionUtils.findMethod;
+
+public class LinkedResourceMethodHandlerTest {
+
+	private final ResourceContent resourceContent = new ResourceContent();
+	private final LinkedResourceMethodHandler handler = createHandler();
+
+	@Test
+	public void supportsLinkedResourceSetterIsTrue() {
+		assertThat(handler.supports(findMethod(ResourceContent.class, "setSomething", String.class)), is(true));
+	}
+
+	@Test
+	public void supportsLinkedResourceGetterIsTrue() {
+		assertThat(handler.supports(findMethod(ResourceContent.class, "getSomething")), is(true));
+	}
+
+	@Test
+	public void supportsNonAnnotatedSetterIsFalse() {
+		assertThat(handler.supports(findMethod(ResourceContent.class, "setSomethingElse", String.class)), is(false));
+	}
+
+	@Test
+	public void supportsNonAnnotatedGetterIsFalse() {
+		assertThat(handler.supports(findMethod(ResourceContent.class, "getSomethingElse")), is(false));
+	}
+
+	@Test
+	public void invokeSetsAndReturnsSameLinkedResource() throws InvocationTargetException, IllegalAccessException {
+		final Method setterMethod = findMethod(ResourceContent.class, "setSomething", String.class);
```

Comment:
Please could you make this test a bit simpler? Aim for one assertion per test if reasonably practicable. I don't think all those intermediate assertions add anything.

```java
handler.invoke(resourceContent, setterMethod, null, new String[]{"x"});

assertThat(handler.invoke(resourceContent, getterMethod, null, null), is("x"));
```

is nearly all you need.